tests/src/main/scala/scalaz/tests/AMaybeTests.scala

```diff
@@ -14,29 +14,31 @@ object AMaybeTests {
   def tests[T](harness: Harness[T]): T = {
     import harness._
     section(
-      namedSection("debug instance")(
-        test("AEmpty") { () =>
-          assert(empty[(?, ?), Int].debugs === "AEmpty")
-        },
-        test("AJust") { () =>
-          assert(just[(?, ?), Int, Int]((0, 1)).debugs === "AJust((0, 1))")
+      namedSection("instances")(
+        namedSection("debug")(
+          test("AEmpty") { () =>
+            assert(empty[(?, ?), Int].debugs === "AEmpty")
+          },
+          test("AJust") { () =>
+            assert(just[(?, ?), Int, Int]((0, 1)).debugs === "AJust((0, 1))")
+          },
+        ),
+        test("eq") { () =>
+          val testData =
+            IList(
+              empty[(?, ?), Int],
+              just[(?, ?), Int, Int]((0, 1)),
+              just[(?, ?), Int, Int]((2, 3)),
+            )
+          testData.cross(testData).foldMap {
+            case (a, b) =>
+              assert((a === b) == (a eq b))
```

Comment:
so equality on `AMaybe` is by reference? that almost seems like it shouldn't have equality at all, or that its equality should depend on some property of the relation parameter.

Reply:
I think this test is meant to be read as "the elements _in this case_ are all distinct, so two elements should only be equal if they are the same element". It would be better if this test more explicitly expressed that only pairs on the diagonal of the cross product are equal.

Reply:
How would I express that directly?

Reply:
`zipWithIndex` before `cross`, and then

```scala
case ((i, a), (j, b)) =>
  assert((a === b) == (i == j))
```

Reply:
Fixed.